packages/plugin-page-view-tracking-browser/src/page-view-tracking.ts

```diff
@@ -20,7 +20,12 @@ export const pageViewTrackingPlugin: CreatePageViewTrackingPlugin = (options: Op
   let localConfig: BrowserConfig;
 
   const createPageViewEvent = async (): Promise<Event> => {
+    const isEventInNewSession = isNewSession(localConfig.sessionTimeout, localConfig.lastEventTime);
+    if (isEventInNewSession && amplitude) {
+      amplitude.setSessionId(Date.now());
```

Comment:
We now have another place where session id is getting reset? Isn't this also being done in the timeline when a new event is fired?

Reply:
We call the setSessionId at [here](https://github.com/amplitude/Amplitude-TypeScript/blob/main/packages/analytics-browser/src/browser-client.ts#L273).
We actually hijack the event while processing and set the sessionId. However, page view event at that time already set the pageCounter. We need to set the session first before firing the page view event.

Reply:
In a plugin, in the execute part is where we enrich an event right? Should the page counter be set in the execute method instead? THat way, the session id is already reset and the page counter is set correctly. 

Setting session id here is more of a fix for the page counter and not something that we are doing because thats the right thing to do. 

cc: @justin-fiedler for feedback as well.

Reply:
It's hard to catch the page view event in the plugin, because the event type can be customized for the page view event. 
And every event will go through any enrichment plugin. That means [here](https://github.com/amplitude/Amplitude-TypeScript/blob/main/packages/plugin-page-view-tracking-browser/src/page-view-tracking.ts#L108-L120) is not only for the page view events.

Reply:
Do we need to `setSessionId()` here? Would it be enough to check if this event is in a new session and clear the page counter? And leave the session management to the core SDK when it processes the page view event?

Reply:
@justin-fiedler  then we need to remove clear page counter logic from the setSessionId function. For me, it makes more sense to have this logic while setting the session because this is related behavior in a new session. We also call setSessioId in web attribution plugins though.

Reply:
There are 2 places where we create a new session
1. When the time between sessions > session timeout. This happens in the browser-client execute method.
2. When UTM parameters are changed within a session, we force create a new session. This happens in the web attribution plugin.

There is no use case where the page view tracking plugin needs to forcefully reset the session id. It seems like we are only calling setSessionId here to indirectly reset the page counter which isn't all that obvious. 

Wrt the event name - its a part of the options object and the options is accessible in the execute method. So can we check the event name against that option and use that to populate page counter? 

Open to what @justin-fiedler is suggesting as well. 

Session management is already fairly complex and adding one more place where session can be reset is just exposing one more possibility of bugs coming up.

Reply:
See [comment](https://github.com/amplitude/Amplitude-TypeScript/pull/720#discussion_r1568314094).